### PR TITLE
BOAC-2027 Allow refresh when Vue needs a loaded user profile

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.ets.berkeley.edu/jira/browse/BOAC-2027

I also have a more focused, Search-only, version of the fix, but @johncrossman thought it might be nice to hook it into the router. What do you think, @lyttam & John C.?

### CODE REVIEW

**Comment by raydavis** [comment]
The Git diff for the router change is hard to follow -- probably best to look at the whole file.

**Comment by johncrossman** [comment]
Looks good to me, @raydavis  Welcome to the realm of typescript/Vue.

**Comment by lyttam** [comment]
I like this approach better than using a watch in Loading. Thanks @raydavis!
